tcl/auth.tcl

```diff
@@ -76,7 +76,7 @@ proc qc::auth_session { session_id } {
     #| Try to authenticate user based on the session_id given
     #| Return the user_id if successful
     #| On failure throw AUTH error
-    if { [session_exists $session_id] } {
+    if { [session_valid $session_id] } {
         return [session_user_id $session_id]
```

Comment:
I've also just realised something odd with `qc::auth` unless I am misunderstanding the intentions:

```tcl
# Try session based auth
    if { [qc::cookie_exists session_id] } {
        set session_id [cookie_get session_id]
        if { [qc::session_exists $session_id] } {
            qc::session_update $session_id
            return [set current_user_id [qc::auth_session $session_id]]
        }
    }
```

It looks to me like it would update the session and then check it which would mean that the idle timeout for a session would never fail because the time-modified was just updated and won't reflect the actual idle time.

I think it should be:

```tcl
# Try session based auth
    if { [qc::cookie_exists session_id] } {
        set session_id [cookie_get session_id]
        if { [qc::session_exists $session_id] } {
            set current_user_id [qc::auth_session $session_id]
            qc::session_update $session_id
            return $current_user_id
        }
    }
```